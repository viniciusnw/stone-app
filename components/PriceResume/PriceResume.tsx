import React from 'react';
import styled from 'styled-components/native';
import { View, Text } from 'react-native';

const StyledContainer = styled(View)``;

const StyledLabel = styled(Text)`
  color: ${({ theme }) => theme.color.black};
  font-weight: 700;
  font-size: 16px;
`;

const StyledDescription = styled(Text)`
  color: ${({ theme }) => theme.color.gray};
  margin-top: 2px;
`;

const PriceResume = ({ label, description, ...props }: any) => {
  return !label && !description ? null : (
    <StyledContainer {...props}>
      {label && <StyledLabel>{label}</StyledLabel>}
      {description && <StyledDescription>{description}</StyledDescription>}
    </StyledContainer>
  );
};

export default PriceResume;
