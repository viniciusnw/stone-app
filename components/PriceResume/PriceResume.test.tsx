import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import PriceResume from "./PriceResume";

jest.useFakeTimers();

test("PriceResume", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <PriceResume label="label" description="description" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
