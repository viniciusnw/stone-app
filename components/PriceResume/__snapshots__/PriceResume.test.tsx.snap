// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceResume 1`] = `
<View
  style={{}}
>
  <Text
    style={
      {
        "color": "#111111",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    label
  </Text>
  <Text
    style={
      {
        "color": "#666666",
        "marginTop": 2,
      }
    }
  >
    description
  </Text>
</View>
`;
